### PR TITLE
[Arch] [TD] Bug 4392: ArchSection and hidden objects for Render mode

### PR DESCRIPTION
**Pull request by Vanuan**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [x] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [x] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

Forum thread:
https://forum.freecadweb.org/viewtopic.php?f=35&t=48263

### CODE REVIEW

**Comment by Vanuan** [comment]
Rebased and fixed the Coin render modes

File: src/Mod/Arch/ArchSectionPlane.py [hunk 1]
```diff
@@ -705,28 +722,28 @@ def getCoinSVG(cutplane,objs,cameradata=None,linewidth=0.2,singleface=False,face
     lset = coin.SoIndexedLineSet()
     lset.coordIndex.setValues(0,2,[0,1])
     sep.addChild(lset)
-    rn.insertChild(sep,0)
+    rootNode.insertChild(sep,0)
 
     # set scenegraph
-    vv.setSceneGraph(rn)
+    inventorView.setSceneGraph(rootNode)
```
Comment (vocx-fc) [inline src/Mod/Arch/ArchSectionPlane.py:728]: We are trying to use a more Pythonic style according to our coding standards, which means using `snake_case` variables. I know most of the code is not like this at the moment, but we do try to change things little by little.

Comment (Vanuan) [reply src/Mod/Arch/ArchSectionPlane.py]: @vocx-fc done

**Comment by Vanuan** [comment]
@yorikvanhavre @carlopav any input?

**Comment by Vanuan** [comment]
@vocx-fc It it all good?

**Comment by carlopav** [comment]
The code looks more readable and tidy, that's good!
Waiting to have a bit of time to test it: I took a weekend off at the lake ;)

**Comment by yorikvanhavre** [comment]
Looks good, thanks!

File: src/Mod/Arch/ArchSectionPlane.py [hunk 2]
```diff
@@ -364,40 +388,25 @@ def getSVG(source,
...
+    should_update_svg_cache = False
+    if not svgcache:
+        should_update_svg_cache = True
 
     # generating SVG
     if renderMode in ["Coin",2,"Coin mono",3]:
         # render using a coin viewer
         if hasattr(source.ViewObject,"ViewData") and source.ViewObject.ViewData:
-            cameradata = getCameraData(source.ViewObject.ViewData)
+            cameradata = None#getCameraData(source.ViewObject.ViewData)
```
Comment (Vanuan) [inline src/Mod/Arch/ArchSectionPlane.py:400]: Hm, not sure why I did that

**Pull request merged**